### ISSUE TITLE
Add `__init__` docstrings to main class page

### ISSUE THREAD

**Issue by jeremyliweishih** [alteryx/evalml#2392]
Currently on the main page for each class we can only see the `__init__` [parameters but not their respective docstrings.](https://evalml.alteryx.com/en/stable/generated/evalml.automl.AutoMLSearch.html#evalml.automl.AutoMLSearch) To do, one needs to click on the `__init__` method [itself](https://evalml.alteryx.com/en/stable/generated/methods/evalml.automl.AutoMLSearch.__init__.html#evalml.automl.AutoMLSearch.__init__). This can be improved by adding it to the main page like [sklearn does.](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.ShuffleSplit.html)

**Comment by dsherry** [comment]
SGTM

FWIW the docstring for AutoMLSearch init and class are identical, "Automated Pipeline search". Let's also add more description there!
